client/src/components/Listing.jsx

```diff
@@ -1,19 +1,37 @@
+
+import axios from 'axios';
 import React from 'react';
 import Photos from './Photos';
 import Details from './Details';
 
 class Listing extends React.Component {
   constructor(props) {
     super(props);
-    this.state = {};
+    this.state = {
+      listings: [],
+    };
+  }
+
+  componentDidMount() {
+    // provide 12 random listings
+    axios.get('/api/rooms/')
```

Comment:
@VALE46GP This endpoint appears to be formatted incorrectly if adhering to RESTful routing principles. Might I suggest `/api/rooms/:roomId/similarListings`

Reply:
@VALE46GP The new endpoint looks much better!